### PR TITLE
Move to plotly

### PR DESCRIPTION
**Pull request by timmens**
This PR extends #288.

In this PR we

- Remove matplotlib and seaborn from the dependencies
- Update the use of plotly figures in the documentation notebooks
- Update the ```derivative_plot```

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -54,4 +54,4 @@ dependencies:
       - DFO-LS
       - fides==0.7.4
       - gif
-      - gif[matplotlib]
```
Comment (janosg) [inline environment.yml:57]: I guess in the dev environment we still need to keep seaborn and gif[matplotlib] to generate some of the plots in the documentation. Can you confirm by running pytask once in the docs folder?

File: src/estimagic/visualization/colors.py [hunk 2]
```diff
@@ -1,4 +1,11 @@
-import seaborn as sns
```
Comment (janosg) [inline src/estimagic/visualization/colors.py:1]: This whole module should go. Are we still using it anywhere?

File: src/estimagic/visualization/convergence_plot.py [hunk 3]
```diff
@@ -30,6 +24,9 @@ def convergence_plot(
     stopping_criterion="y",
     x_precision=1e-4,
     y_precision=1e-4,
+    combine_plots_in_grid=True,
+    template=PLOTLY_TEMPLATE,
+    palette=px.colors.qualitative.Plotly,
```
Comment (janosg) [inline src/estimagic/visualization/convergence_plot.py:29]: For the color palette we should also have a global (imported from config) as default. I.e. similar to `PLOTLY_TEMPLATE`.

While we are at it: I think the colors look really bad in derivative plots and not too good in some others. Maybe we should try out D3, G10 and T10 from [here](https://plotly.com/python/discrete-color/) to see if we can get a nicer default without much effort.

**Review by janosg — APPROVED** [review]

File: environment.yml (hunk 1 above)
Comment (timmens) [reply environment.yml:57]: You're right. For the documentation we need both seaborn and matplotlib.

File: src/estimagic/visualization/colors.py (hunk 2 above)
Comment (timmens) [reply src/estimagic/visualization/colors.py:1]: Moved to ``src/dashboard``.

File: src/estimagic/visualization/convergence_plot.py (hunk 3 above)
Comment (timmens) [reply src/estimagic/visualization/convergence_plot.py:29]: The Set2 template provides a slightly better default, but prospectively we may want to create our own.

Comment (hmgaudecker) [reply src/estimagic/visualization/convergence_plot.py:29]: My suggestion: Let's not get distracted. Economists should trust specialization. If anything, pick some color cycle from somewhere, but let's not take on the burden of discussions about this sort of thing.

**Pull request merged**